doc/man3/ASYNC_start_job.pod

```diff
@@ -112,7 +112,14 @@ with them. Applications can wait for the file descriptor to be ready for "read"
 using a system function call such as select or poll (being ready for "read"
 indicates that the job should be resumed). If no file descriptor is made
 available then an application will have to periodically "poll" the job by
-attempting to restart it to see if it is ready to continue.
+attempting to restart it to see if it is ready to continue. From OpenSSL 3.0.0,
+an addtional callback and data pointers would be stored within the
+ASYNC_WAIT_CTX along with an additional status field that can be used for the
+notification of retries from an engine. The intention would be that an engine
+could be written to look whether the callback has been set. If it has then it
+would use the addtional mechanism in preference to the file descriptor
+notifications. If a callback is not set then the behavior would be identical to
+that before. See ASYNC_WAIT_CTX_new() for more details.
```

Comment:
I'd rewrite this to say something like "Applications can be notified in two different ways that a job is ready to be resumed". Then, in separate paragraphs briefly describe the two approaches. Remove the "From OpenSSL 3.0.0" reference - these are the man pages for the 3.0.0 version. Everything is relevant to that version. Users of older versions should look at the older man pages.